code/datums/extensions/extension_statmod.dm

```diff
@@ -50,52 +50,84 @@ STATMOD_LAYER	=	list(/datum/proc/reset_layer)
 
 //Trigger all relevant update procs without changing registration
 /datum/extension/proc/update_statmods()
-	var/mob/M = holder
```

Comment:
okay statmods shouldn't just be for mods, genericising them is fine, but not these specific typechecks, just make it work for all datums

Reply:
all the stuff you've touched in statmods is going to need work, i recommend reverting this file and encumbrance, and start over